File: ch3/locus.py
order = [ int(x) for x in list(input("Order-based: ")) ]
locus = [ 0 for i in range(len(order)) ]

for j in range(len(order)):
    try:
        locus[order[j]] = order[j + 1]
    except IndexError:
        locus[order[j]] = order[0]

print(f"Locus-based: {''.join(map(str, locus))}")

File: ch3/locus.java
import java.util.Arrays;
import java.util.Scanner;
import java.util.stream.Stream;

public class locus{
    public static void main(String[] args){
        final int LENGTH = 10;
        String order_s;
        Scanner sc = new Scanner(System.in);

        System.out.print("Order-based: ");
        order_s = sc.next();

        int[] order = Stream.of(order_s.split("")).mapToInt(Integer::parseInt).toArray();
        int[] locus = new int[order.length];

        for (int i = 0; i < LENGTH; i++){
            try{
                locus[order[i]] = order[i + 1];
            }catch(IndexOutOfBoundsException e){
                locus[order[i]] = order[0];
            }
        }

        System.out.println("Locus-based: " + Arrays.toString(locus).replaceAll("[^0-9]",""));
    }
}
